pandas/core/reshape/pivot.py

```diff
@@ -431,17 +431,31 @@ def _convert_by(by):
 def pivot(data: "DataFrame", index=None, columns=None, values=None) -> "DataFrame":
     if columns is None:
         raise TypeError("pivot() missing 1 required argument: 'columns'")
+    columns = columns if is_list_like(columns) else [columns]
 
     if values is None:
-        cols = [columns] if index is None else [index, columns]
+        cols: List[str] = []
```

Comment:
I think you want `List[Label]` (import from pandas._typing) to allow anything column name, unless this really does have to be string-only

Reply:
i don't think this will matter until we type the signature, so ok with handling this as a followon.